projects/com.oracle.truffle.llvm.nodes/src/com/oracle/truffle/llvm/nodes/intrinsics/llvm/x86/LLVMX86_64BitVAStart.java

```diff
@@ -108,73 +126,124 @@ private FrameSlot getStackPointerSlot() {
+                        } else {
+                            assert overflowAreaSize >= overflowOffset;
+                            overflowOffset += storeArgument(overflowArgArea + overflowOffset, object);
+                        }
+                        break;
+
+                    case OVERFLOW_AREA:
+                        assert overflowAreaSize >= overflowOffset;
+                        overflowOffset += storeArgument(overflowArgArea + overflowOffset, object);
+                        break;
+
+                    default:
+                        CompilerDirectives.transferToInterpreter();
+                        throw new IllegalStateException("not supported: " + area);
                 }
             }
         }
 
         return null;
     }
 
+    private static int calculateUsedGpArea(Object[] realArguments, int numberOfExplicitArguments) {
+        assert numberOfExplicitArguments <= realArguments.length;
+
+        int usedGpArea = 0;
+        for (int i = 0; i < numberOfExplicitArguments && usedGpArea < X86_64BitVarArgs.GP_LIMIT; i++) {
+            if (getVarArgArea(realArguments[i]) == VarArgArea.GP_AREA) {
+                usedGpArea += X86_64BitVarArgs.GP_STEP;
+            }
+        }
+
+        return usedGpArea;
+    }
+
+    private static int calculateUsedFpArea(Object[] realArguments, int numberOfExplicitArguments) {
+        assert numberOfExplicitArguments <= realArguments.length;
+
+        int usedFpArea = 0;
+        final int fpAreaLimit = X86_64BitVarArgs.FP_LIMIT - X86_64BitVarArgs.GP_LIMIT;
+        for (int i = 0; i < numberOfExplicitArguments && usedFpArea < fpAreaLimit; i++) {
+            if (getVarArgArea(realArguments[i]) == VarArgArea.FP_AREA) {
+                usedFpArea += X86_64BitVarArgs.FP_STEP;
+            }
+        }
+
+        return usedFpArea;
+    }
+
+    private static int calculateOverflowArea(Object[] realArguments, int numberOfExplicitArguments) {
+        assert numberOfExplicitArguments <= realArguments.length;
+
+        int overflowArea = 0;
+        int gpOffset = calculateUsedGpArea(realArguments, numberOfExplicitArguments);
+        int fpOffset = X86_64BitVarArgs.GP_LIMIT + calculateUsedFpArea(realArguments, numberOfExplicitArguments);
+        for (int i = numberOfExplicitArguments; i < realArguments.length; i++) {
+            VarArgArea area = getVarArgArea(realArguments[i]);
+            if (area == VarArgArea.GP_AREA && gpOffset < X86_64BitVarArgs.GP_LIMIT) {
+                gpOffset += X86_64BitVarArgs.GP_STEP;
+            } else if (area == VarArgArea.FP_AREA && fpOffset < X86_64BitVarArgs.FP_LIMIT) {
+                fpOffset += X86_64BitVarArgs.FP_STEP;
+            } else if (area != VarArgArea.OVERFLOW_AREA) {
+                overflowArea += X86_64BitVarArgs.STACK_STEP;
+            } else {
+                // TODO: we need to know the required stacksize
+                overflowArea += 128;
```

Comment:
oh, that's a workaround before the time I knew the stacksize. I will fix it immediatly